hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/DynamicConfigTemplate.java

```diff
@@ -355,7 +357,8 @@ void addMapConfig(String name, int backupCount, int asyncBackupCount, int timeTo
                       @Nullable String partitioningStrategyClassName,
                       @Nullable Data partitioningStrategyImplementation,
                       @Nullable HotRestartConfig hotRestartConfig,
-                      @Since("1.6") int mergeBatchSize);
+                      @Since("1.6") int mergeBatchSize,
+                      @Since("1.8") String preprocessing);
```

Comment:
In the doc, it mentions as if it can have only two values, CREATION_TIME or OFF, which seems to be an enumeration. We can better represent enumerations with Integer type.

Reply:
I used InMemoryFormat and EvictionPolicy as example. They are also encoded as strings.

Reply:
I think such uses were overseen during reviews. We always prefer enumeration for enumeration when possible (e.g. auth response code, etc.)